apis/r/R/QueryCondition.R

```diff
@@ -0,0 +1,316 @@
+                value = switch(
+                    arrow_type_name,
+                    ascii = rhs_text,
+                    utf8 = rhs_text,
+                    bool = as.logical(rhs_text),
+                    date32 = as.POSIXct(rhs_text),
+                    timestamp = as.Date(rhs_text),
+                    as.numeric(rhs_text)),
+                arrow_type_name = arrow_type_name,
+                op_name = .map_op_to_character(op_name),
+                qc = tiledbsoma_empty_query_condition(somactx)))
+
+        } else {
+            stop("Unexpected token in expression: ", format(node))
+        }
+    }
+
+    # Use base-r `substitute` to map the user-provided expression to a parse tree
+    parse_tree <- substitute(expr)
+
+    # Map the parse tree to TileDB core QueryCondition
+    return(.parse_tree_to_qc(parse_tree, debug))
+}
+
+# ================================================================
+#' An S4 class for a TileDB QueryCondition object
+#'
+#' @slot ptr An external pointer to the underlying implementation
+#' @slot init A logical variable tracking if the query condition object has been
+#' initialized
+#' @exportClass tiledbsoma_query_condition
+setClass(
+    "tiledbsoma_query_condition",
+    slots = list(ptr = "externalptr", init = "logical"))
+
+# ================================================================
+#' Creates a 'tiledbsoma_query_condition' object
+#'
+#' @param somactx (optional) A TileDB Ctx object; if not supplied the default
+#' context object is retrieved
+#' @return A 'tiledbsoma_query_condition' object
+#' @export
+tiledbsoma_empty_query_condition <- function(somactx) {
+    stopifnot("The argument must be a somactx object" = is(somactx, "externalptr"))
+    ptr <- libtiledbsoma_empty_query_condition(somactx)
+    query_condition <- new("tiledbsoma_query_condition", ptr = ptr, init = FALSE)
+    invisible(query_condition)
+}
+
+# ================================================================
+#' Initialize a 'tiledbsoma_query_condition' object
+#'
+#' Initializes (and possibly allocates) a query condition object using a triplet of
+#' attribute name, comparison value, and operator.  Six types of conditions are supported,
+#' they all take a single scalar comparison argument and attribute to compare against.
+#' At present only integer or numeric attribute comparisons are implemented.
+#' @param attr_name A character value with the scheme attribute name
+#' @param value A scalar value that the attribute is compared against
+#' @param arrow_type_name A character value with the TileDB data type of the attribute column, for
+#' example 'float' or 'int32'
+#' @param op_name A character value with the comparison operation. This must be one of
+#' 'LT', 'LE', 'GT', 'GE', 'EQ', 'NE'.
+#' @param qc A 'tiledbsoma_query_condition' object to be initialized by this call.
+#' @return The initialized 'tiledbsoma_query_condition' object
+#' @export
+tiledbsoma_query_condition_from_triple <- function(
```

Comment:
Keep this internal to make it easier to re-build/re-write at a later date
```suggestion
#' @return The initialized 'tiledbsoma_query_condition' object
#' 
#' @noRd
#'
tiledbsoma_query_condition_from_triple <- function(
```